### PR TITLE
STY: use pytest.raises context syntax

### PR DESCRIPTION
**Pull request by jbrockmendel**
There are still about a thousand non-context usages of pytest.raises.

@gfyoung if there is a subset of these that especially merit match kwargs, let me know.  I'm not eager to manually go through all of them, but doing something systematic or manually doing a few I'd be up for.

### CODE REVIEW

File: pandas/tests/frame/test_indexing.py [hunk 1]
```diff
@@ -3459,35 +3457,30 @@ def test_assigning_ops(self):
...
-
         #   - assign a complete row (mixed values) -> exp_single_row
         df = orig.copy()
         df.iloc[2, :] = ["b", 2]
         tm.assert_frame_equal(df, exp_single_row)
 
         #   - assign a complete row (mixed values) not in categories set
-        def f():
+        with pytest.raises(ValueError):
+            np.log(s)
```
Comment (jreback) [inline pandas/tests/frame/test_indexing.py]: what's this

**Review by jreback — CHANGES_REQUESTED** [review]
can you create a lint rule to check for non-context manager usage of ``pytest.raises``?

File: pandas/tests/frame/test_indexing.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_indexing.py]: must be a copy/paste mixup, will fix

**Comment by gfyoung** [comment]
@jbrockmendel : Thanks for the ping.  Given how massive this is, I think we can just focus on the conversion to context manager first.  Which ones should use a match parameter can be determined after (i.e. subsequent PR’s)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks. @gfyoung we should prob have a lint rule to detect this, but leave commented for now

**Comment by gfyoung** [comment]
@jreback: I actually was discussing the checking if error messages, not the usage of the context manager versus not.  That being said, I do agree we should lint for what was addressed here.

**Pull request merged**